src/Sentry.Unity.Native/SentryNative.cs

```diff
@@ -75,12 +81,37 @@ public static void Configure(SentryUnityOptions options, ISentryUnityInfo sentry
                 crashedLastRun = SentryNativeBridge.HandleCrashedLastRun(options);
                 PerDirectoryCrashInfo.Add(cacheDirectory, crashedLastRun);
 
-                options.DiagnosticLogger?
+                Logger?
                     .LogDebug("Native SDK reported: 'crashedLastRun': '{0}'", crashedLastRun);
             }
         }
         options.CrashedLastRun = () => crashedLastRun;
+
+        ShouldReinstallBackend = true;
     }
 
-    public static void ReinstallBackend() => SentryNativeBridge.ReinstallBackend();
+    // We're calling this in `BeforeSceneLoad` instead of `SubsystemRegistration` as it's too soon and the
+    // SignalHandler would still get overwritten.
+    [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.BeforeSceneLoad)]
+    private static void ReinstallBackend()
+    {
+        // The backend should only be reinstalled if the native SDK has been initialized successfully.
+        if (!ShouldReinstallBackend)
+        {
+            Logger?.LogDebug("Skipping reinstalling the native backend as the initialization seems to have failed.");
+            return;
+        }
```

Comment:
how about we also log if we do this, since I suspect we might want to debug this:

```suggestion
        } else {
            Logger?.LogDebug("Reinstalling the native backend to make sure we capture native crashes.");
        }
```